core/src/tile/tileManager.cpp

```diff
@@ -384,11 +388,24 @@ void TileManager::loadSubTasks(std::vector<std::shared_ptr<DataSource>>& _subSou
         if (it != subTasks.end() && (*it)->subTaskId() == int(index)) { continue; }
 
         TileID subTileID = tileID;
+
+        // get tile for lower zoom if we are past max zoom
         if (subTileID.z > subSource->maxZoom()) {
             subTileID = subTileID.withMaxSourceZoom(subSource->maxZoom());
         }
         auto subTask = subSource->createTask(subTileID, index);
-        if (subTask->isReady()) {
+        // check if we are at valid zoom for source
+        if (!subSource->isActiveForZoom((float)subTileID.z)) {
+            // Right now all subSources are raster, but let's check anyway...
+            if (RasterSource* rasterSubSource = dynamic_cast<RasterSource*>(subSource.get())) {
+                subTasks.insert(it, subTask);
+                // If the subSource isn't active for the zoom, we should just
+                // load an empty texture and move on.
+                rasterSubSource->loadEmptyTexture(std::move(subTask));
```

Comment:
Seems like what you may want to do in that case is attach a raster source with a lower **`max_zoom`** value, and the raster texture will be scaled up at the higher zooms, whereas the `max_display_zoom` parameter was meant to really signify you *don't* want this tile source rendering at a higher zoom level. In our current styles like Walkabout, we set the terrain normals `max_zoom: 15`, and let them be scaled up onto vectors up to z20.

I don't know what is happening with the current behavior (@tallytalwar can illuminate), but it may be difficult to ensure correct drawing in all cases where a raster is attached. We have to provide *some* texture to the GL shader, and with placeholder like a 1x1 pixel image, I would not expect the lighting normal calculations to be correct. You'd want different default textures/behavior for different types of raster attachments (e.g. treating the raster as a direct color texture, vs. treating as a normal map), and there is also the "custom" raster mode where you don't know the intentions of the user.

Am I missing something in the example above, where only the `max_zoom` needs to be set?